### PR TITLE
ETS-33 Adds a basic UI for Transfers, including integration tests

### PR DESCRIPTION
**Pull request by matt-bernhardt**
** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/ETS-33

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/views/transfer/new.html.erb
Comment (JPrevost) [inline app/views/transfer/new.html.erb:4]: 🙈 (and yes, I know this is how we do this already in this app so it is totally appropriate but also 🙈 )

Comment (matt-bernhardt) [reply app/views/transfer/new.html.erb:4]: Heh - I've been faced with this exact problem in some personal projects, and noted that this is how it happens here. From a design perspective, it would be interesting to discuss a better way of doing this at some point.

But yes, a bit of 🙈 for now.

File: app/views/transfer/new.html.erb
Comment (JPrevost) [inline app/views/transfer/new.html.erb:13]: I think this should be the default for this field, but for `theses_admins` I believe it should just display _all_ Departments. If that is a pain to make that change (and associated tests) I would be okay with opening a specific ticket to address that distinction. The current solution will technically not restrict admins from submitting, but the form will never allow them to choose anything for this field so effectively they still can't submit :)

Comment (matt-bernhardt) [reply app/views/transfer/new.html.erb:13]: Hmm... ok. I had thought that, for users with the `:thesis_admin` ability, this call would end up returning all departments because their ability would be for all departments. If we want to structure this differently, that's fine - I can make this change and its test.

Comment (JPrevost) [reply app/views/transfer/new.html.erb:13]: The way it's modeled we need to handle that in the UI here or would need more complex behavior in the models. I normally move towards complexity in the models so maybe we should talk about your expectation as it may be a very good one.

**Review by JPrevost — CHANGES_REQUESTED** [review]
If we can roll in the change to allow thesis_admins to submit to any dept I think that would be great to include, but if it gets messy we can open a new ticket.

I'd like to hold off on merging until we get stakeholder feedback if we can.

**Comment by matt-bernhardt** [comment]
I'm working on the change to how thesis_admins see the list of available departments, but this will likely to take until Monday at this point - what should be a very short change has turned into 5-7 failing tests. I'm blaming Friday PM brain.

I agree with getting stakeholder approval, so I think this buys me time to work out this last bit.

File: app/views/transfer/new.html.erb
Comment (matt-bernhardt) [reply app/views/transfer/new.html.erb]: I think this is resolved now...

**Review by JPrevost — APPROVED** [review]
🌈

**Pull request merged**
